Update docs with the new changes (#15)
wso2-extensions/identity-conditional-auth-typingdna: docs/config.md

> To use TypingDNA with WSO2 Identity Server, first you need to configure the authenticator with WSO2 Identity Server. The following topics provide instructions on how to configure the TypingDNA with WSO2 Identity Server as a risk-based authentication (RBA) option:

Reviewer: General comment: Images are not aligned with the numbering.
<img width="1114" alt="Screenshot 2021-11-12 at 13 14 58" src="https://user-images.githubusercontent.com/21237558/141429535-8fb5ff36-7fb8-4287-bb74-e576f7026696.png">

Author: Addressed with https://github.com/wso2-extensions/identity-conditional-auth-typingdna/pull/15/commits/809fb935c083c4e9bed561daf6bd729580a9e074

```
Note: TypingDNA is supported with WSO2 Identity Server 5.12.0-alpha10 version onwards.
```

> * [Enable TypingDNA in the WSO2 Identity Server](#enable-typingdna-in-the-wso2-identity-server)

Reviewer: ```suggestion
* [Enabling TypingDNA in the WSO2 Identity Server](#enable-typingdna-in-the-wso2-identity-server)
```

* [Deploying TypingDNA Artifacts](#deploying-typingdna-artifacts)
* [Setting up the TypingDNA account](#setting-up-the-typingdna-account)
* [Configuring the TypingDNA in WSO2 Identity Server](#configuring-the-typingdna-in-wso2-identity-server)
* [Configuring the application to use TypingDNA](#configuring-the-application-to-use-typingdna)

> ### Enable TypingDNA in the WSO2 Identity Server

Reviewer: ```suggestion
### Enabling TypingDNA in the WSO2 Identity Server
```

    enabled_features=["security.loginVerifyData.typingDNA"]
   ```

### Deploying TypingDNA Artifacts

You can either download the TypingDNA artifacts or build the authenticator from the source code.

> 

Reviewer: - To build from the source code:
- To build the authenticator, navigate to the `identity-conditional-auth-typingdna` directory and execute the following command in a command prompt:
- Copy the `org.wso2.carbon.identity.conditional.auth.typingdna.functions-x.x.x.jar` file into the 
   `<IS-Home>/repository/components/dropins` directory and the `api#identity#typingdna#v_.war` file into the `<IS-Home>/repository/deployment/server/webapps` directory.

Author: Addressed with https://github.com/wso2-extensions/identity-conditional-auth-typingdna/pull/15/commits/809fb935c083c4e9bed561daf6bd729580a9e074

### Configuring the TypingDNA in WSO2 Identity Server

1. Start the WSO2 Identity Server and log in to the management console using admin credentials.

> 2. Go to `Identity Providers -> Resident -> Other settings -> TypingDNA Configuration`

Reviewer: ```suggestion
2. Go to `Identity Providers -> Resident -> Other settings -> TypingDNA Configuration`.
```

> 4. Enable TypingDNA & configure API Key, Secret. You can get the Key & Secret from TypingDNA

Reviewer: ```suggestion
4. Enable TypingDNA and configure the **API key** and **secret**. You can get the key and secret from the TypingDNA [dashboard](https://www.typingdna.com/clients/).
```

>    [dashboard](https://www.typingdna.com/clients/).

Reviewer: ```suggestion
```

>    Refer [this doc](files/Sign%20In.pdf) for detailed information.

Reviewer: ```suggestion
   [Learn more](files/Sign%20In.pdf).
```

### Configuring the application to use TypingDNA

> 1. Go to `Service Providers -> List` & Select the sample application you have configured and click `Edit`.

Reviewer: ```suggestion
1. Go to `Service Providers -> List`, select the sample application you have configured, and click `Edit`.
```

2. Expand `Local and Outbound Authentication Configuration` and click `Advanced Configuration`.

> 3. Configure the required authentication for two steps and use the TypingDNA adaptive script as below.

Reviewer: ```suggestion
3. Configure two authentication steps (2FA) in the login flow and use the Typing DNA adaptive script as shown below.
```

```
// This script will step up 2FA authentication if the user's typing behaviour mis-match with enrolled behaviour.

// You can use score(num 0-100), result(boolean), confidence(num 0-100), comparedPatterns in your logic to promote 2nd step
// here result is used at typingVerified.result

var onLoginRequest = function(context) {
    executeStep(1, {
        onSuccess: function (context) {
            verifyUserWithTypingDNA(context, {
                onSuccess: function(context,data){
                    // Change the definition here if you want.
                    var userVerified = data.result;

                    // data.isTypingPatternReceived indicates whether a typing patterns is received from login portal.
                    if (data.isTypingPatternReceived && !userVerified){
                        executeStep(2);
                    }
                },onFail: function(context,data){
                    executeStep(2);
                }

            });
        }
    });
};

// End of TypingDNA-Based.......
```
![Alt_text](images/screen-shot-4.png?raw=true)
![Alt_text](images/screen-shot-6.png?raw=true)
![Alt_text](images/screen-shot-7.png?raw=true)

>  Refer [this doc](files/adaptive-script-description.md) to get detailed information about TypingDNA adaptive functions.

Reviewer: ```suggestion
 Learn more about [TypingDNA adaptive functions](files/adaptive-script-description.md).
```
